### ISSUE TITLE
KeyError exception when running await bath.get() with a synchronous context manager

### ISSUE THREAD

**Issue by lwhist** [numat/huber#24]
When running

```
async def get():
    with Bath('192.168.10.04') as bath:
        print(await bath.get())

asyncio.run(get())
```

I get the following error:

Exception has occurred: KeyError
'writer'
  File "test.py", line 19, in get
    print(await tempBath.get())
          ^^^^^^^^^^^^^^^^^^^^
  File "test.py", line 21, in <module>
    asyncio.run(get())
KeyError: 'writer'

**Comment by alexrudd2** [comment]
Again, thank you for the report!  I think you found a problem in either the code or the readme.

The important line is `    with Bath('192.168.10.04') as bath:`
The keyword `with` is what's called a "[context manager](https://realpython.com/python-with-statement/#using-the-python-with-statement)".  

It will call the special method `__enter__()`.  Notice that this code does not call `self._connect()`, which means that `connection['writer']` is not created, which is I think causing your `KeyError`.  (You didn't do anything wrong, I believe the readme example doesn't work)

https://github.com/numat/huber/blob/05f8f155d4dd98f46278fff96581632e95d4a42d/huber/driver.py#L46-L48

It's also possible to use an async context manager  `async with`.  This will call `__aenter__()` instead, which works.

https://github.com/numat/huber/blob/05f8f155d4dd98f46278fff96581632e95d4a42d/huber/driver.py#L50-L56

**Comment by alexrudd2** [comment]
I'm not sure if it's even possible to fix the sync context manager.  Can you try the following instead?

```python
async def get():
    async with Bath('192.168.10.04') as bath:    # NOTICE async here!
        print(await bath.get())

asyncio.run(get())
```

**Comment by alexrudd2** [comment]
P.S. To get pretty colors
![Screenshot 2023-04-25 at 11 07 24 AM](https://user-images.githubusercontent.com/52292902/234337167-6cec4844-85d6-426b-a3a1-3e1a9024cf8b.png)

**Comment by lwhist** [comment]
I am glad to help :)

Looking at it again tomorrow, I had to quit for today.

**Comment by lwhist** [comment]
Adding `async` before `with` works nicely. :)

```python
 async with Bath('192.168.10.4') as bath:
```
Just a sidetrack: I noticed that the IP-address as I previously wrote with a `0` before the `4` does not work (leading 0s are not removed).

**Comment by alexrudd2** [comment]
> Adding `async` before `with` works nicely. :)

Great, thanks!  I've fixed the readme in 67f7a847862910801e4aa22658def0b8f19637cf

> Just a sidetrack: I noticed that the IP-address as I previously wrote with a `0` before the `4` does not work (leading 0s are not removed).

Yeah, the code below passes the IP along as a string to [`asyncio.open_connection`](https://docs.python.org/3/library/asyncio-stream.html#asyncio.open_connection).  It's probably trying to interpret the leading 0 as an octal number instead of a decimal.  (StackOverflow [explanation](https://superuser.com/a/857787))
https://github.com/numat/huber/blob/67f7a847862910801e4aa22658def0b8f19637cf/huber/driver.py#L158

I'm not going to change anything; as you've already discovered removing the leading zero works.
